File: CondoManager.Domain/Entities/Visitor.cs
using CondoManager.Domain.Enums;

namespace CondoManager.Domain.Entities;

public class Visitor
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public string Name { get; set; }
    public string CPF { get; set; }
    public VisitorStatus Status { get; set; }
    public DateOnly DateOfVisit { get; set; }
    public Guid AuthorizedById { get; set; }
    public Resident AuthorizedBy { get; set; }
}

File: frontend/src/src/types/visitor.ts
export type VisitorStatus = 'Pending' | 'Approved' | 'Denied' | 'CheckedIn' | 'CheckedOut'

export interface Visitor {
    id: string
    name: string
    cpf: string
    status: VisitorStatus
    dateOfVisit: string
    authorizedById: string
    condominiumId: string
}

export interface CreateVisitorDto {
    name: string
    cpf: string
    dateOfVisit: string
    authorizedById: string
    condominiumId: string
}